packages/eslint-plugin/src/rules/prefer-literal-enum-member.ts

```diff
@@ -59,16 +59,46 @@ export default createRule({
           return;
         }
 
+        // Allow bitwise operators when both operands are literals or refer to another member on this enum
         if (
           allowBitwiseExpressions &&
           node.initializer.type === AST_NODE_TYPES.BinaryExpression &&
           ['|', '&', '^', '<<', '>>', '>>>'].includes(
             node.initializer.operator,
           ) &&
-          node.initializer.left.type === AST_NODE_TYPES.Literal &&
-          node.initializer.right.type === AST_NODE_TYPES.Literal
+          node.parent.type === AST_NODE_TYPES.TSEnumDeclaration
         ) {
-          return;
+          const leftIsLiteral =
+            node.initializer.left.type === AST_NODE_TYPES.Literal;
+          const rightIsLiteral =
+            node.initializer.right.type === AST_NODE_TYPES.Literal;
+
+          // early branch, so gathering member identifiers only happens when needed
+          if (leftIsLiteral && rightIsLiteral) {
+            return;
+          }
+
+          const members = node.parent.members.reduce((names, member) => {
+            if ('name' in member.id) {
+              names.push(member.id.name);
+            }
+            return names;
+          }, new Array<string>());
```

Comment:
[Style] Heh, I'm surprised this didn't trigger any enabled lint rules. `.reduce<string[]>( ..., [])` would be our preferred syntax here.